fedimint-core/src/db/mod.rs

```diff
@@ -216,6 +223,80 @@ impl Database {
                 .expect("db autocommit retry counter overflowed");
         } // end of loop
     }
+
+    /// Waits for key to be notified.
+    ///
+    /// Calls the `f` on every notifications. Return the value when `f` returns
+    /// a `Some(T)`
+    pub async fn wait_key_check<'a, K, T>(
+        &'a self,
+        key: &K,
+        f: impl Fn(Option<K::Value>) -> Result<Option<T>>,
+    ) -> Result<(T, DatabaseTransaction<'a>)>
+    where
+        K: DatabaseKey + DatabaseRecord,
+    {
+        // not compile time yet :/
+        assert!(
+            <K as DatabaseKey>::NOTIFY,
+            "wait_key called with key({}) without NOTIFY flag",
+            std::any::type_name::<K>()
+        );
+        let key_bytes = if let Some(module_id) = self.module_instance_id {
+            let mut prefix_bytes = vec![MODULE_GLOBAL_PREFIX];
+            module_id
+                .consensus_encode(&mut prefix_bytes)
+                .expect("Error encoding module instance id as prefix");
+            prefix_bytes.extend(key.to_bytes());
```

Comment:
Can we make this a function? `IsolatedDatabaseTransaction` uses very similar logic